internal/querynodev2/segments/reducer.go

```diff
@@ -0,0 +1,33 @@
+package segments
+
+import (
+	"context"
+
+	"github.com/milvus-io/milvus/internal/proto/segcorepb"
+
+	"github.com/milvus-io/milvus-proto/go-api/schemapb"
+	"github.com/milvus-io/milvus/internal/proto/internalpb"
+	"github.com/milvus-io/milvus/internal/proto/querypb"
+)
+
+type internalReducer interface {
+	Reduce([]*internalpb.RetrieveResults) (*internalpb.RetrieveResults, error)
+}
+
+func CreateInternalReducer(ctx context.Context, req *querypb.QueryRequest, schema *schemapb.CollectionSchema) internalReducer {
+	if req.GetReq().GetIsCount() {
+		return &cntReducer{}
+	}
+	return newDefaultLimitReducer(ctx, req, schema)
+}
+
+type segCoreReducer interface {
+	Reduce([]*segcorepb.RetrieveResults) (*segcorepb.RetrieveResults, error)
+}
+
+func CreateSegCoreReducer(ctx context.Context, req *querypb.QueryRequest, schema *schemapb.CollectionSchema) segCoreReducer {
+	if req.GetReq().GetIsCount() {
+		return &cntReducerSegCore{}
+	}
+	return newDefaultLimitReducerSegcore(ctx, req, schema)
+}
```

Comment:
NIT: maybe change the interface to have ctx parameter instead of store ctx in reducer struct
Also, could we combine these two interface into one?

Reply:
It may be implemented with a generic interface

Reply:
I'll move the ctx as the parameter of reducer.